store/sqlstore/channel_store.go

```diff
@@ -521,9 +481,16 @@ func (s SqlChannelStore) createInitialSidebarCategoriesT(transaction *gorp.Trans
 	}
 
 	if !hasCategoryOfType(model.SidebarCategoryFavorites) {
```

Comment:
nit: a `hasCategoryOfType` that is simply a `map[model.SidebarCategoryType]bool` would be more idiomatic than looping every time.